### PR TITLE
Fix #5108 - quote call to ExpandObjects for paths with spaces

### LINKED ISSUE #5108: Internal Logging Issue with OpenStudio 3.7 when it is Installed in a Path with a Space (eg. C:\Program Files\openstudio)
Issue overview
--------------
I am opening this issue primarily for @jmarrec after [asking an initial question on another issue](https://github.com/NREL/OpenStudio/issues/5069#issuecomment-1986642054). The issue is that, whenever OpenStudio is installed in a file path with a space (for example, the `C:\Program Files` directory that IT departments make us use), we get the following message when running an OSW with a reporting measure using OpenStudio CLI:

```
'C:/Program' is not recognized as an internal or external command,
operable program or batch file.
RunEnergyPlus: Completed Successfully with 0 Fatal Errors, 0 Severe Errors, 34 Warnings.
```

## Current Behavior
The message does not indicate a complete OpenStudio CLI failure since the OSW run still completes successfully, including a successful run of EnergyPlus and the creation of reports from reporting measures specified in the OSW. However, it seems that whatever part of the OpenStudio 3.7 code that is producing this exception prevents us from seeing the progress of the reporting measure, whether it's the log of steps running in the measure or any adapters specified in the OSW, like those that allow us to see the progress of the EnergyPlus simulation as the OSW is being run.

## Expected Behavior
In OpenStudio 3.6.1, we did not get this message about "'C:/Program' is not recognized as an internal or external command," and we would instead see the logs of the reporting measure being run as well as the adapter we use giving us a full report of the EnergyPlus simulation as it was run. For example, at a minimum, we would see the following:

```
ExpandObjects Started.
No expanded file generated.
ExpandObjects Finished with Error(s). Time:     0.016
RunEnergyPlus: Completed Successfully with 0 Fatal Errors, 0 Severe Errors, 34 Warnings.
```

... and, if we had an adapter specified in the OSW, we would see the full EnergyPlus progress.

## Steps to Reproduce
I know that @jmarrec is not a Windows user so I think he may be able to create the issue on his end just by installing OpenStudio in a path with a space in it and trying to run the [OpenStudio Results measure](https://github.com/NREL/openstudio-common-measures-gem/tree/develop/lib/measures/openstudio_results) on any model (using the `/path with spaces/openstudio/bin/openstudio` to call the OpenStudio CLI executable). With this, I think he should be able to see the exception message and at least figure out where it is coming from with the OpenStudio code without the need to do a whole Windows setup. Also, he should not need to set up an adapter in the OSW or do anything fancy just to see this initial problem. The issue appears to be more deeply related to the ability to log the progress of reporting measures and not specific to the adapter as I originally thought.

If anyone else wants to recreate the issue or if my suggestion above does not work, I have a minimal OSW file that will work on Windows as long as the OpenStudio executable is installed in `C:\Program Files\openstudio\bin\openstudio.exe` (or another path with a space in it). Here is the zip file of the sample:
https://drive.google.com/file/d/16Rb6gnid0JOxPlcLd78du7F0A5R96x3J/view?usp=sharing

And, if you extract it, this is what you should see:
image

You have a small shoe box OSM, a Boston EPW and a workflow.osw that is meant to run only the OpenStudio Results measure, all out of a C:\test_osm` directory. Here are the contents of that simple OSW JSON:
```
{
    "seed_file": "C:\\test_os\\in.osm",
    "weather_file": "C:\\test_os\\USA_MA_Boston-Logan.Intl.AP.725090_TMY3.epw", 
    "steps": [
        {
            "arguments": {
                "schedules_overview_section": true, 
                "space_type_breakdown_section": true, 
                "plant_loops_detail_section": true, 
                "measure_warning_section": true, 
                "plug_loads_section": true, 
                "zone_equipment_detail_section": true, 
                "outdoor_air_section": true, 
                "air_loops_detail_section": true, 
                "envelope_section_section": true, 
                "space_type_details_section": true, 
                "source_energy_section": true, 
                "monthly_overview_section": true, 
                "water_use_section": true, 
                "interior_lighting_section": true, 
                "hvac_load_profile": true, 
                "building_summary_section": true, 
                "annual_overview_section": true, 
                "utility_bills_rates_section": true, 
                "units": "IP", 
                "reg_monthly_details": false, 
                "exterior_light_section": true, 
                "cost_summary_section": true, 
                "zone_summary_section": true, 
                "zone_condition_section": true
            }, 
            "measure_dir_name": "openstudio_results"
        }
    ]
}
```
Just run the OSW using:
```
cd C:\test_os
"C:\Program Files\openstudio\bin\openstudio.exe" run -w workflow.osw
```
... and you should see the issue.

## Possible Solution
I have a feeling that all that is needed to fix this issue is that the path to a file or folder needs to be put in "double quotes" somewhere in the OpenStudio code. This way, the full path will be interpreted by the command line and it won't try to interpret 'C:/Program' as a command.

### Environment

I'm running on Windows 10 but, as I said, I think any OpenStudio installation with a space in the path should illustrate the issue.

### Context
This is not a deal breaker for us upgrading to OpenStudio 3.7 since, as I said, the OSW files are still running and producing the correct outputs. However, several of our users are starting to complain that "OpenStudio is frozen" since they can't see any logging or progress of the reporting measure. [Here is an example](https://discourse.pollination.cloud/t/energy-simulation-issue-with-ver-po-v1-43-8-0/3699).

Thanks, as always.

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------

 - Fixes #5108 
 - Automated testing is too tricky to bother, but I did test before and after manually

```
ln -sf /Applications/EnergyPlus-23-2-0 "energy plus2320"
export ENERGYPLUS_EXE_PATH="$(pwd)/energy plus2320/energyplus"
openstudio run --style-stdout --show-stdout -w workflow.osw
```

Before:

```
sh: /Users/julien/Downloads/temp/5108/energy: No such file or directory
[openstudio.workflow.OSWorkflow] <Warn> ExpandObjects returned a non-zero exit code (32512).
```

After:

`$ "/Users/julien/Downloads/temp/test os380fix/bin/openstudio" run --style-stdout --show-stdout -w workflow.osw`

```
ExpandObjects Started.
No expanded file generated.
ExpandObjects Finished. Time:     0.014
```

### Pull Request Author

 - [ ] Model API Changes / Additions
 - [ ] Any new or modified fields have been implemented in the EnergyPlus ForwardTranslator (and ReverseTranslator as appropriate)
 - [ ] Model API methods are tested (in `src/model/test`)
 - [ ] EnergyPlus ForwardTranslator Tests (in `src/energyplus/Test`)
 - [ ] If a new object or method, added a test in NREL/OpenStudio-resources: Add Link
 - [ ] If needed, added VersionTranslation rules for the objects (`src/osversion/VersionTranslator.cpp`)
 - [ ] Verified that C# bindings built fine on Windows, partial classes used as needed, etc.
 - [ ] All new and existing tests passes
 - [ ] If methods have been deprecated, update rest of code to use the new methods

**Labels:**

 - [ ] If change to an IDD file, add the label `IDDChange`
 - [ ] If breaking existing API, add the label `APIChange`
 - [ ] If deemed ready, add label `Pull Request - Ready for CI` so that CI builds your PR

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] Code Style, strip trailing whitespace, etc.
 - [ ] All related changes have been implemented: model changes, model tests, FT changes, FT tests, VersionTranslation, OS App
 - [ ] Labeling is ok
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified

### CODE REVIEW

File: src/workflow/RunEnergyPlus.cpp [hunk 1]
```diff
@@ -157,7 +157,7 @@ void OSWorkflow::runEnergyPlus() {
     // TODO: workflow-gem was manually running expandObjects prior to the potential serialization to json
     // Should we rather pass -x to the E+ cmd line?
     if (!workflowJSON.runOptions()->skipExpandObjects()) {
-      const std::string cmd = openstudio::toString(runDirResults.expandObjectsExe.native());
+      const std::string cmd = fmt::format("\"{}\"", openstudio::toString(runDirResults.expandObjectsExe.native()));
```
Comment (jmarrec) [inline src/workflow/RunEnergyPlus.cpp:160]: expandObjectsExe is infered from getEnergyPlusDirectory which uses getEnergyPlusExecutable which will use ENERGYPLUS_EXE_PATH (as long as it's not running from the build dir)

**Comment by jmarrec** [comment]
@chriswmackey binary is above

**Comment by chriswmackey** [comment]
Thanks, @jmarrec .
Clicking on the .exe doesn't seem to start the download for me. Is there something that I'm missing here?

**Comment by chriswmackey** [comment]
Nevermind. I think it was just a bothersome security feature in Chrome that was not letting my download executables from addresses that didn't start with https. I have the exe now.

**Pull request merged**